slidechain/cmd/export/export.go

```diff
@@ -51,38 +50,29 @@ func main() {
 	}
 	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
 	defer cancel()
-	var (
-		asset        xdr.Asset
-		exportAmount int64
-		inputAmount  int64
-		err          error
-	)
+
+	var err error
+	asset := stellar.NativeAsset()
 	if *code != "" {
 		asset, err = stellar.NewAsset(*code, *issuer)
 		if err != nil {
 			log.Fatalf("error creating asset from code %s and issuer %s: %s", *code, *issuer, err)
 		}
-		exportAmount, err = strconv.ParseInt(*amount, 10, 64)
-		if err != nil {
-			log.Fatalf("error parsing export amount %s: %s", *amount, err)
-		}
-		inputAmount, err = strconv.ParseInt(*input, 10, 64)
-		if err != nil {
-			log.Fatalf("error parsing input amount %s: %s", *input, err)
-		}
-	} else {
-		asset = stellar.NativeAsset()
-		exportXlm, err := xlm.Parse(*amount)
-		if err != nil {
-			log.Fatalf("error parsing export amount %s: %s", *amount, err)
-		}
-		exportAmount = int64(exportXlm)
-		inputXlm, err := xlm.Parse(*input)
-		if err != nil {
-			log.Fatalf("error parsing input amount %s: %s", *input, err)
-		}
-		inputAmount = int64(inputXlm)
 	}
+
+	// XDR scales down an amount unit of every asset by a factor of 10^7.
+	// Thus, xlm.Parse works for both native and non-native assets.
+	exportXlm, err := xlm.Parse(*amount)
```

Comment:
Suggest calling this `exportAmount` (since it's not XLM-specific) and, instead of creating a new variable to hold the `int64`-typed version of this value, just type-convert inline at the point you use it.

Same thing for `inputXlm` below.